sdk/samples/core/mosquitto_rpc_client_sample.c

```diff
@@ -0,0 +1,297 @@
+      {
+        if (recv_data->status != AZ_MQTT5_RPC_STATUS_OK)
+        {
+          printf(
+              LOG_APP_ERROR "Error response received. Status :%d. Message :%s\n",
+              recv_data->status,
+              az_span_ptr(recv_data->error_message));
+        }
+        else
+        {
+          if (!az_span_is_content_equal(content_type, recv_data->content_type))
+          {
+            printf(
+                LOG_APP_ERROR "Invalid content type. Expected: {%s} Actual: {%s}\n",
+                az_span_ptr(content_type),
+                az_span_ptr(recv_data->content_type));
+          }
+          else
+          {
+            // TODO: deserialize
+            printf(
+                LOG_APP "Command response received: %s\n",
+                az_span_ptr(recv_data->response_payload));
+          }
+        }
+        ret = remove_command(&pending_commands, recv_data->correlation_id);
+      }
+      else
+      {
+        printf(
+            LOG_APP_ERROR "Request with correlation id: %s not found\n",
+            az_span_ptr(recv_data->correlation_id));
+      }
+      (void)ret;
+      break;
+    }
+
+    case AZ_HFSM_EVENT_ERROR:
+      printf(LOG_APP_ERROR "Error Event\n");
+      // az_platform_critical_error();
+      break;
+
+    default:
+      // TODO:
+      break;
+  }
+
+  return AZ_OK;
+}
+
+int main(int argc, char* argv[])
+{
+  (void)argc;
+  (void)argv;
+
+  /* Required before calling other mosquitto functions */
+  if (mosquitto_lib_init() != MOSQ_ERR_SUCCESS)
+  {
+    printf(LOG_APP "Failed to initialize MosquittoLib\n");
+    return -1;
+  }
+
+  printf(LOG_APP "Using MosquittoLib %d\n", mosquitto_lib_version(NULL, NULL, NULL));
+
+  az_log_set_message_callback(az_sdk_log_callback);
+  az_log_set_classification_filter_callback(az_sdk_log_filter_callback);
+
+  connection_context = az_context_create_with_expiration(
+      &az_context_application, az_context_get_expiration(&az_context_application));
+
+  az_mqtt5 mqtt5;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_init(&mqtt5, NULL));
+
+  az_mqtt5_x509_client_certificate primary_credential = (az_mqtt5_x509_client_certificate){
+    .cert = cert_path1,
+    .key = key_path1,
+  };
+
+  az_mqtt5_connection_options connection_options = az_mqtt5_connection_options_default();
+  connection_options.client_id_buffer = client_id;
+  connection_options.username_buffer = username;
+  connection_options.password_buffer = AZ_SPAN_EMPTY;
+  connection_options.hostname = hostname;
+  connection_options.client_certificates[0] = primary_credential;
+
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_init(
+      &mqtt_connection, &connection_context, &mqtt5, iot_callback, &connection_options));
+
+  az_mqtt5_property_bag property_bag;
+  mosquitto_property* mosq_prop = NULL;
+  LOG_AND_EXIT_IF_FAILED(az_mqtt5_property_bag_init(&property_bag, &mqtt5, &mosq_prop));
+
+  LOG_AND_EXIT_IF_FAILED(az_rpc_client_hfsm_init(
```

Comment:
Nit: naming - the application shouldn't need to understand `hfsm`. (low priority).
Maybe we call this one "client" (and corresponding stateful "server") and we can call the other layer something like "codec" or "protocol" layer.

Reply:
Done - we decided to call this the az_rpc_client_policy and leave az_rpc_client as is since the policy adds extra functionality to the client